internal/clients/http/ec2/ec2_client.go

```diff
@@ -442,6 +485,52 @@ func getRoleName(arn string) (string, error) {
 	return roleName[1], nil
 }
 
+func getStatementFromAWSDocument(document string) ([]Statement, error) {
+	unescapedDoc, err := url.QueryUnescape(document)
+	if err != nil {
+		return nil, fmt.Errorf("Could not parse url from the Document: %w", err)
+	}
+
+	var policyVersion Version
+	err = json.Unmarshal([]byte(unescapedDoc), &policyVersion)
+	if err != nil {
+		return nil, fmt.Errorf("Could not unmarshal json: %w", err)
+	}
+	return policyVersion.Statement, nil
+}
+
+func listStatements(versions []*iamTypes.PolicyVersion) ([]Statement, error) {
+	var result []Statement
```

Comment:
Do we want `result` to be null if empty or an empty array? 
https://stackoverflow.com/questions/25543520/declare-slice-or-make-slice

Reply:
Good point. I have decided to use a slice in this case rather than an array. AWS does not really have the best way of returning policy Document. We have to decode the url and unmarshal the json. Therefore we do not know the length of the resulting slice/array before we decode and unmarshal it.
In this case I thought it is more convenient to use a slice, as it is more dynamic and does not require memory allocation for each append.

Reply:
I totally agree that we should use slices, I was referring to what is the correct way to initialize it. 
I was asking if we prefer `result` to be null if empty or if we want it to be an empty slice (that is the discussion I have pasted in the comment above)